lib/plan-finder.js

```diff
@@ -152,22 +157,37 @@ class PlanFinder {
                 * @param {String} name transformer name
                 * @param {Object} attributes custom attributes for this step
      */
-    formPlan(source, outputInstructions, currentPlan) {
+    async formPlan(source, outputInstructions, currentPlan) {
         if (!currentPlan || currentPlan.length === 0) {
             debug(`No plan could be found for : ${source.type} -> ${outputInstructions.type}`);
             throw new RenditionFormatUnsupportedError(`No plan could be found for : ${source.type} -> ${outputInstructions.type}`);
         }
+
         // userData is an important field for sending IO events in Nui
         // for now, add them to every output step instructions
         // future plan: store these somewhere, maybe in the plan or engine level to put inside every event
         const userData = outputInstructions.userData;
-
         const plan = [];
+
+        // get image metadata and check for "Orientation" property if this is for Sensei
+        if (this.registeredTransformers.length === 1 && this.registeredTransformers[0].includes("SenseiTransformer")) {
```

Comment:
So the idea is we look at the case where there is exactly only one transformer registered, and check if it is for Sensei? I don't really understand why the length must be `1`? 
We already have more than one Sensei transformer registered, normally (eg. in worker-flite or worker-pie, you can see it in the logs when the registration happens). They are in a catalog, and each manifest leads to a transformer instance, see here: https://git.corp.adobe.com/nui/transformer-sensei/blob/master/lib/transformer-catalog.js#L34-L47.

Reply:
this should be looking at `currentPlan.length` not the list of registered transformers